src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -441,6 +441,19 @@ public override bool IsDefined(Type attributeType, bool inherit)
     }
 """;
 
+    public static string AntiforgeryMetadataType = """
+file sealed class AntiforgeryMetadata : IAntiforgeryMetadata
+{
+    public static IAntiforgeryMetadata ValidationRequired = new AntiforgeryMetadata(true);
+
+    public AntiforgeryMetadata(bool requiresValidation)
```

Comment:
Could you use primary constructor here too?

Reply:
I thought about it but in my mind I established an (in hindsight arbitrary) benchmark that the generated code should be compatible with C#11. I suppose there is nothing stopping us from using C#12 language features but I'd be more keen about doing https://github.com/dotnet/aspnetcore/issues/46496 in that case.